Items Editing/simon_Kontakt Sample Namer.lua

```diff
@@ -0,0 +1,209 @@
+	set_color = true
+	retval, retvals_csv = reaper.GetUserInputs("Sample Export Namer", 5, "Number of Dynamic Levels,Number of Round Robins,Read Pitch from Regions?,Import track name?,Color dynamic levels?",tostring(dyn_amount) .. "," ..tostring(rr_amount) .. "," ..tostring(import_pitch) .. "," ..tostring(import_track) .. "," ..tostring(set_color))
+
+	if retval == true then -- if user clicked ok
+		dyn_amount, rr_amount, import_pitch, import_track, set_color = retvals_csv:match("([^,]+),([^,]+),([^,]+),([^,]+),([^,]+)")
+		if dyn_amount then dyn_amount = tonumber(dyn_amount) end
+		if dyn_amount == 0 then dyn_amount = 1 end
+
+		if rr_amount then rr_amount = tonumber(rr_amount) end
+		if rr_amount == 0 then rr_amount = 1 end
+
+		if import_pitch == "true" then import_pitch = true else import_pitch = false end
+		if import_track == "true" then import_track = true else import_track = false end
+		if set_color == "true" then set_color = true else set_color = false end
+	end
+	return retval
+end
+
+function AppendRR()
+	count_sel_item = reaper.CountSelectedMediaItems(0)
+	count_sel_tracks = reaper.CountSelectedTracks(0)
+	  
+	i = 0 -- goes from 0 to selected number of items
+	j = 0 -- goes from 0 to number of tracks
+	k = 1 -- goes from 1 to number of RRs (j is the number in "RR1", "RR2", etc)
+
+	for j = 0, count_sel_tracks -1 do -- Runs through selected tracks one by one
+
+
+		for k = 1, rr_amount do -- runs through rr number one by one
+
+			while i < count_sel_item/count_sel_tracks do -- runs through media items by multiples of rr_amount
+				
+				item = reaper.GetSelectedMediaItem(0, i)
+				item_take = reaper.GetActiveTake(item)
+
+				local _, name = reaper.GetSetMediaItemTakeInfo_String(item_take, "P_NAME", "" .. k, false)
+				local _, name = reaper.GetSetMediaItemTakeInfo_String(item_take, "P_NAME", name .. "_RR_" .. k, true)
+
+				
+				i = i + rr_amount
+			end
+			i = k
+		end
+	end
+end
+
+function AppendDyn()
+	count_sel_item = reaper.CountSelectedMediaItems(0)
+	count_sel_tracks = reaper.CountSelectedTracks(0)
+	dyn_size = math.floor(127 / dyn_amount)
+	  
+	i = 0 -- goes from 0 to selected number of items
+	j = 0 -- goes from 0 to number of tracks
+	k = 1 -- goes from 1 to number of dynamics (j is the number in "RR1", "RR2", etc)
+	l = 0 -- corresponds to dyn number
+
+	for j = 0, count_sel_tracks -1 do -- Runs through selected tracks one by one
+
+		i = 0
+		while i < count_sel_item/count_sel_tracks do -- runs through items on one track by multiples of rr_amount
+
+			for l = 1, dyn_amount do -- l corresponds to current dynamic level
+
+				if l == 1 then -- initialize dynamic numbers for each new note group
+					dyn_min = 1
+					dyn_max = dyn_size
+				end
+
+				if l == dyn_amount then
+					dyn_max = 127
+				end
+
+				for k = 0, rr_amount - 1 do -- k loops through sets of rrs and applies dynamic level text to items
+					item = reaper.GetSelectedMediaItem(0, i + k)
+					item_take = reaper.GetActiveTake(item)
```

Comment:
`item` can be nil here (eg. one item on one selected track, `i + k` is out of bounds) causing a crash on this line.